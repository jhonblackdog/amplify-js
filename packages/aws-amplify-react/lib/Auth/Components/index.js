"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var WithAuthSignIn_1 = require("./WithAuthSignIn");
exports.WithAuthSignIn = WithAuthSignIn_1.default;
//# sourceMappingURL=index.js.map