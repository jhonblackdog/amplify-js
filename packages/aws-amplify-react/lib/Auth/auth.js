"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = y[op[0] & 2 ? "return" : op[0] ? "throw" : "next"]) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [0, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
Object.defineProperty(exports, "__esModule", { value: true });
var AWS = require("aws-sdk/global");
var cognito_auth_1 = require("./cognito-auth");
var ErrorUtils_1 = require("../Core/ErrorUtils");
/**
* Provide authentication steps
*/
var Auth = (function () {
    function Auth(options) {
        var awsUserPoolsId = options.awsUserPoolsId, awsUserPoolsWebClientId = options.awsUserPoolsWebClientId, region = options.region, awsCognitoIdentityPoolId = options.awsCognitoIdentityPoolId;
        if (!awsUserPoolsId) {
            ErrorUtils_1.default.missingParameter('awsUserPoolsId');
        }
        if (!awsUserPoolsWebClientId) {
            ErrorUtils_1.default.missingParameter('awsUserPoolsWebClientId');
        }
        if (!awsCognitoIdentityPoolId) {
            ErrorUtils_1.default.missingParameter('awsCognitoIdentityPoolId');
        }
        this.authOptions = {
            awsUserPoolsId: awsUserPoolsId,
            awsUserPoolsWebClientId: awsUserPoolsWebClientId,
            region: region || AWS.config.region,
            awsCognitoIdentityPoolId: awsCognitoIdentityPoolId
        };
        this.cognitoAuth = new cognito_auth_1.default(this.authOptions);
    }
    //TODO: Currently we are returning Promise<**CognitoInterface**> We should make sure this is want we want to return. We may want a middle layer that is either more helpful or more generic so we are not tied to Cognito.
    Auth.prototype.signUp = function (params) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                console.log("params for signup on library side ", params);
                return [2 /*return*/, this.cognitoAuth.handleNewCustomerRegistration(params.username, params.password, params.SignupAttributes)];
            });
        });
    };
    Auth.prototype.confirmSignUp = function (username, verificationCode) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                console.log("username" + username);
                return [2 /*return*/, this.cognitoAuth.handleSubmitVerificationCode(username, verificationCode)];
            });
        });
    };
    Auth.prototype.resendSignUpMFA = function (username) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                return [2 /*return*/, this.cognitoAuth.handleResendVerificationCode(username)];
            });
        });
    };
    Auth.prototype.signIn = function (username, password) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                console.log("2: us" + typeof (username) + "pass" + typeof (password));
                console.log("username: " + username + " password: " + password);
                return [2 /*return*/, this.cognitoAuth.handleSignIn(username, password)];
            });
        });
    };
    Auth.prototype.submitSignInMFA = function (user, verificationCode) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                console.log("MFA code and username : ", verificationCode);
                console.log("user is ::: " + user);
                //user = JSON.parse(user);
                console.log("cog user : ", JSON.stringify(user));
                return [2 /*return*/, this.cognitoAuth.sendMFAVerificationCode(user, verificationCode)];
            });
        });
    };
    Auth.prototype.signOut = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                return [2 /*return*/, this.cognitoAuth.handleSignOut()];
            });
        });
    };
    Auth.prototype.forgotPassword = function (username) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                return [2 /*return*/, this.cognitoAuth.handleForgotPassword(username)];
            });
        });
    };
    Auth.prototype.confirmMFAForgotPassword = function (username, verificationCode, newPassword) {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                return [2 /*return*/, this.cognitoAuth.handleForgotPasswordReset(username, verificationCode, newPassword)];
            });
        });
    };
    Auth.prototype.getCredWithThirdParty = function (token, thirdPartyProvider) {
        this.cognitoAuth.getCredWithThirdParty(token, thirdPartyProvider);
    };
    return Auth;
}());
exports.Auth = Auth;
//# sourceMappingURL=auth.js.map