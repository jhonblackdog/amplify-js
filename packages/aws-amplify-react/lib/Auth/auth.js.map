{"version":3,"file":"auth.js","sourceRoot":"","sources":["../../src/Auth/auth.ts"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;AAAA,oCAAsC;AACtC,+CAAwD;AAGxD,iDAA4C;AAG5C;;EAEE;AACF;IAIE,cAAY,OAAoB;QACtB,IAAA,uCAAc,EAAE,yDAAuB,EAAE,uBAAM,EAAE,2DAAwB,CAAa;QAE9F,EAAE,CAAC,CAAC,CAAC,cAAc,CAAC,CAAC,CAAC;YACpB,oBAAU,CAAC,gBAAgB,CAAC,gBAAgB,CAAC,CAAC;QAChD,CAAC;QACD,EAAE,CAAC,CAAC,CAAC,uBAAuB,CAAC,CAAC,CAAC;YAC7B,oBAAU,CAAC,gBAAgB,CAAC,yBAAyB,CAAC,CAAC;QACzD,CAAC;QACD,EAAE,CAAC,CAAC,CAAC,wBAAwB,CAAC,CAAC,CAAC;YAC9B,oBAAU,CAAC,gBAAgB,CAAC,0BAA0B,CAAC,CAAC;QAC1D,CAAC;QAED,IAAI,CAAC,WAAW,GAAG;YACjB,cAAc,EAAE,cAAc;YAC9B,uBAAuB,EAAE,uBAAuB;YAChD,MAAM,EAAE,MAAM,IAAI,GAAG,CAAC,MAAM,CAAC,MAAM;YACnC,wBAAwB,EAAE,wBAAwB;SACnD,CAAC;QAEF,IAAI,CAAC,WAAW,GAAG,IAAI,sBAAW,CAAC,IAAI,CAAC,WAAW,CAAC,CAAC;IACvD,CAAC;IAED,0NAA0N;IAC7M,qBAAM,GAAnB,UAAoB,MAAoB;;;gBACtC,OAAO,CAAC,GAAG,CAAC,oCAAoC,EAAE,MAAM,CAAC,CAAC;gBAC1D,sBAAO,IAAI,CAAC,WAAW,CAAC,6BAA6B,CAAC,MAAM,CAAC,QAAQ,EAAE,MAAM,CAAC,QAAQ,EAAE,MAAM,CAAC,gBAAgB,CAAC,EAAC;;;KAClH;IAEY,4BAAa,GAA1B,UAA2B,QAAgB,EAAE,gBAAwB;;;gBACnE,OAAO,CAAC,GAAG,CAAC,UAAU,GAAG,QAAQ,CAAC,CAAC;gBACnC,sBAAO,IAAI,CAAC,WAAW,CAAC,4BAA4B,CAAC,QAAQ,EAAE,gBAAgB,CAAC,EAAC;;;KAClF;IAEY,8BAAe,GAA5B,UAA6B,QAAgB;;;gBAC3C,sBAAO,IAAI,CAAC,WAAW,CAAC,4BAA4B,CAAC,QAAQ,CAAC,EAAC;;;KAChE;IAEY,qBAAM,GAAnB,UAAoB,QAAgB,EAAE,QAAgB;;;gBACpD,OAAO,CAAC,GAAG,CAAC,OAAO,GAAG,OAAO,CAAC,QAAQ,CAAC,GAAG,MAAM,GAAG,OAAO,CAAC,QAAQ,CAAC,CAAC,CAAC;gBACtE,OAAO,CAAC,GAAG,CAAC,YAAY,GAAG,QAAQ,GAAG,aAAa,GAAG,QAAQ,CAAC,CAAC;gBAChE,sBAAO,IAAI,CAAC,WAAW,CAAC,YAAY,CAAC,QAAQ,EAAE,QAAQ,CAAC,EAAC;;;KAE1D;IAGY,8BAAe,GAA5B,UAA6B,IAAiB,EAAE,gBAAwB;;;gBACtE,OAAO,CAAC,GAAG,CAAC,0BAA0B,EAAE,gBAAgB,CAAC,CAAC;gBAC1D,OAAO,CAAC,GAAG,CAAC,cAAc,GAAG,IAAI,CAAC,CAAC;gBACnC,0BAA0B;gBAC1B,OAAO,CAAC,GAAG,CAAC,aAAa,EAAE,IAAI,CAAC,SAAS,CAAC,IAAI,CAAC,CAAC,CAAC;gBACjD,sBAAO,IAAI,CAAC,WAAW,CAAC,uBAAuB,CAAC,IAAI,EAAE,gBAAgB,CAAC,EAAC;;;KACzE;IAGY,sBAAO,GAApB;;;gBACE,sBAAO,IAAI,CAAC,WAAW,CAAC,aAAa,EAAE,EAAC;;;KACzC;IAEY,6BAAc,GAA3B,UAA4B,QAAgB;;;gBAC1C,sBAAO,IAAI,CAAC,WAAW,CAAC,oBAAoB,CAAC,QAAQ,CAAC,EAAC;;;KACxD;IAEY,uCAAwB,GAArC,UAAsC,QAAgB,EAAE,gBAAwB,EAAE,WAAmB;;;gBACnG,sBAAO,IAAI,CAAC,WAAW,CAAC,yBAAyB,CAAC,QAAQ,EAAE,gBAAgB,EAAE,WAAW,CAAC,EAAC;;;KAC5F;IAEM,oCAAqB,GAA5B,UAA6B,KAAU,EAAE,kBAA0B;QACjE,IAAI,CAAC,WAAW,CAAC,qBAAqB,CAAC,KAAK,EAAE,kBAAkB,CAAC,CAAA;IACnE,CAAC;IAEH,WAAC;AAAD,CAAC,AA3ED,IA2EC;AA3EY,oBAAI"}