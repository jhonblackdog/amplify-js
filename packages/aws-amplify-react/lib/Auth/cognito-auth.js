"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var amazon_cognito_identity_js_1 = require("amazon-cognito-identity-js");
var AWS = require("aws-sdk");
var CognitoAuth = (function () {
    function CognitoAuth(config) {
        this.config = config;
        this.userPool = new amazon_cognito_identity_js_1.CognitoUserPool({
            UserPoolId: config.awsUserPoolsId,
            ClientId: config.awsUserPoolsWebClientId
        });
    }
    //let cognitoUser: CognitoUser = null;
    /**********************
     * Login methods *
     **********************/
    CognitoAuth.prototype.getSignInCallback = function (cognitoUser, res, rej) {
        var _this = this;
        var signinCallback = {
            onSuccess: function (result) {
                console.log('result: ', result);
                var loginCred = 'cognito-idp.' + _this.config.region + '.amazonaws.com/' + _this.config.awsUserPoolsId;
                var Login = {};
                Login[loginCred] = result.getIdToken().getJwtToken();
                var credJson = {
                    IdentityPoolId: _this.config.awsCognitoIdentityPoolId,
                    Logins: (_a = {},
                        _a[loginCred] = result.getIdToken().getJwtToken(),
                        _a),
                };
                AWS.config.credentials = new AWS.CognitoIdentityCredentials(credJson);
                AWS.config.getCredentials(function (error) {
                    if (error) {
                        console.error(error);
                        return;
                    }
                    var _a = AWS.config.credentials, accessKeyId = _a.accessKeyId, secretAccessKey = _a.secretAccessKey, sessionToken = _a.sessionToken;
                    var awsCredentials = {
                        accessKeyId: accessKeyId,
                        secretAccessKey: secretAccessKey,
                        sessionToken: sessionToken
                    };
                });
                var _a;
            },
            onFailure: function (error) {
                console.log('sign in onFailure' + JSON.stringify(error));
                _this.logHelpfulError(error);
                rej(error);
            },
            mfaRequired: function (challengeName, challengeParameters) {
                console.log('sign in MFA required');
                console.log("cognito user from charlotte: ", cognitoUser);
                res({
                    charlottetype: "mfareq",
                    user: cognitoUser,
                    obj: {
                        challengeName: challengeName, challengeParameters: challengeParameters
                    },
                    sendCode: function (code) {
                        console.log('The code:', code);
                    }
                });
            },
            newPassword: function (userAttributes, requiredAttributes) {
                //TODO: Handle
            },
            customChallenge: function (challengeParameters) {
                //TODO: Handle
            }
        };
        return signinCallback;
    };
    ;
    CognitoAuth.prototype.handleSignIn = function (username, password) {
        var _this = this;
        console.log("in the lib: username: " + username + " password " + password);
        var authenticationDetails = new amazon_cognito_identity_js_1.AuthenticationDetails({
            Username: username,
            Password: password
        });
        console.log("Authdetails: ", authenticationDetails);
        var cognitoUser = new amazon_cognito_identity_js_1.CognitoUser({
            Username: username,
            Pool: this.userPool
        });
        return new Promise(function (res, rej) {
            console.log(typeof (username));
            console.log(typeof (password));
            cognitoUser.authenticateUser(authenticationDetails, _this.getSignInCallback(cognitoUser, res, rej));
        });
    };
    //TODO: Currently not used needed for sign in 
    CognitoAuth.prototype.sendMFAVerificationCode = function (user, code) {
        var _this = this;
        return new Promise(function (res, rej) {
            /*const sendMFACallbacks = {
                onSuccess: (session: CognitoUserSession) => {
                    res(session);
                },
                onFailure: (err: any) => {
                    rej(err);
                }
            }*/
            user.sendMFACode(code, _this.getSignInCallback(user, res, rej));
        });
    };
    /**********************
     * Registration methods *
     **********************/
    CognitoAuth.prototype.logRegistrationError = function (error) {
        var err = error.toString();
        if (/UsernameExistsException: User already exists$/.test(err)) {
            console.log(err);
        }
        else if (/InvalidParameterException?/.test(err) || /InvalidPasswordException?/.test(err)) {
            console.log(err + '. Password may not meet requirements');
        }
    };
    //TODO: This needs to be updated to accept all parameters, not just those the sample uses
    CognitoAuth.prototype.handleNewCustomerRegistration = function (username, password, signUpAttributeList) {
        var _this = this;
        var attributeList = [];
        console.log("in the func : " + username, password, signUpAttributeList);
        console.log("attribute list : ", signUpAttributeList);
        return new Promise(function (res, rej) {
            var signUpCallbacks = function (err, result) {
                if (err) {
                    rej(err);
                }
                else {
                    console.log("Signup successful now returning result ", result);
                    res(result);
                }
            };
            _this.userPool.signUp(username, password, signUpAttributeList, null, signUpCallbacks);
        });
    };
    CognitoAuth.prototype.handleSubmitVerificationCode = function (username, verificationCode) {
        var cognitoUser = new amazon_cognito_identity_js_1.CognitoUser({
            Username: username,
            Pool: this.userPool
        });
        return new Promise(function (res, rej) {
            var verificationCallback = function (err, result) {
                if (err) {
                    rej(err);
                }
                else {
                    res(result);
                }
            };
            cognitoUser.confirmRegistration(verificationCode, true, verificationCallback);
        });
    };
    CognitoAuth.prototype.handleResendVerificationCode = function (username) {
        var cognitoUser = new amazon_cognito_identity_js_1.CognitoUser({
            Username: username,
            Pool: this.userPool
        });
        return new Promise(function (res, rej) {
            var resendVerificationCallback = function (error, result) {
                if (error) {
                    rej(error);
                }
                else {
                    res(result);
                }
            };
            cognitoUser.resendConfirmationCode(resendVerificationCallback);
        });
    };
    /*************************
     * Forgot Password methods *
     *************************/
    CognitoAuth.prototype.checkResetPasswordError = function (error) {
        if ((/UserNotFoundException?/.test(error))
            || (/InvalidParameterException: Cannot reset password for the user as there is no registered?/.test(error))) {
            return { invalidCodeOrPasswordMessage: 'Invalid username' };
        }
        else if (/LimitExceededException: Attempt limit exceeded, please try after some time?/.test(error)) {
            return { invalidCodeOrPasswordMessage: 'Attempt limit exceeded, please try again later' };
        }
        else if (/CodeMismatchException?/.test(error)) {
            return { invalidCodeOrPasswordMessage: 'Invalid Verfication Code' };
        }
        else if (/InvalidParameterException: Cannot reset password for the user as there is no registered\/verified email or phone_number?$/.test(error)) {
            return { invalidCodeOrPasswordMessage: 'Cannot reset password for the user as there is no registered\/verified email or phone_number' };
        }
        else if ((/InvalidParameterException?/.test(error)) || (/InvalidPasswordException?$/.test(error))) {
            return { invalidCodeOrPasswordMessage: 'Password must contain 8 or more characters with atleast one lowercase,uppercase, numerical and special character' };
        }
    };
    CognitoAuth.prototype.handleForgotPassword = function (username) {
        var _this = this;
        var cognitoUser = new amazon_cognito_identity_js_1.CognitoUser({
            Username: username,
            Pool: this.userPool
        });
        return new Promise(function (res, rej) {
            var forgotPasswordCallBack = {
                onSuccess: function () {
                    console.log('Password reset successful');
                    res();
                },
                onFailure: function (err) {
                    console.log(err.toString());
                    //todo : convert the error messages that we get from checkResetPasswordError to log statements
                    var invalidCodeOrPasswordMessage = _this.checkResetPasswordError(err.toString());
                    rej(err);
                },
                inputVerificationCode: function (data) {
                    //resolve with data
                    res(data);
                }
            };
            cognitoUser.forgotPassword(forgotPasswordCallBack);
        });
    };
    CognitoAuth.prototype.handleForgotPasswordReset = function (username, verificationCode, newPassword) {
        var _this = this;
        var cognitoUser = new amazon_cognito_identity_js_1.CognitoUser({
            Username: username,
            Pool: this.userPool
        });
        return new Promise(function (res, rej) {
            var forgotPasswordCallBack = {
                onSuccess: function () {
                    console.log('Password reset successful');
                    res();
                },
                onFailure: function (err) {
                    console.log(err.toString());
                    //todo : convert the error messages that we get from checkResetPasswordError to log statements
                    var invalidCodeOrPasswordMessage = _this.checkResetPasswordError(err.toString());
                    rej(err);
                },
                inputVerificationCode: function (data) {
                    //resolve with data
                    res(data);
                }
            };
            cognitoUser.confirmPassword(verificationCode, newPassword, forgotPasswordCallBack);
        });
    };
    /*****************
     * SignOut methods *
     *****************/
    CognitoAuth.prototype.handleSignOut = function () {
        /*
        const userPool = new CognitoUserPool({
            UserPoolId: awsmobile.aws_user_pools_id, // Your user pool id here
            ClientId: awsmobile.aws_user_pools_web_client_id // Your client id here
        });*/
        var cognitoUser = this.userPool.getCurrentUser();
        cognitoUser.signOut();
    };
    CognitoAuth.prototype.logHelpfulError = function (error) {
        var err = error.toString();
        if (/InvalidParameterException: Missing required parameter USERNAME$/.test(err)) {
            console.log(err);
        }
        else if (/UserNotFoundException?/.test(err)
            || /NotAuthorizedException?/.test(err)) {
            console.log(err + '. Username or password may be invalid.');
        }
        else if (/CodeMismatchException: Invalid code or auth state for the user.$/.test(err)) {
            console.log(err);
        }
        else if (/InvalidParameterException: Missing required parameter SMS_MFA_CODE$/.test(err)) {
            console.log(err + 'Verficiation code can not be empty');
        }
    };
    /*****************
     * Cognito federated identities methods *
     *****************/
    CognitoAuth.prototype.getCredWithThirdParty = function (token, thirdPartyProvider) {
        var Login = { thirdPartyProvider: token };
        var credJson = {
            IdentityPoolId: this.config.awsCognitoIdentityPoolId,
            Logins: Login
        };
        AWS.config.credentials = new AWS.CognitoIdentityCredentials(credJson);
        AWS.config.getCredentials(function (error) {
            if (error) {
                throw new Error('Failed to archive aws credential with the token provided.');
            }
            var _a = AWS.config.credentials, accessKeyId = _a.accessKeyId, secretAccessKey = _a.secretAccessKey, sessionToken = _a.sessionToken;
            var awsCredentials = {
                accessKeyId: accessKeyId,
                secretAccessKey: secretAccessKey,
                sessionToken: sessionToken
            };
        });
    };
    return CognitoAuth;
}());
exports.default = CognitoAuth;
//# sourceMappingURL=cognito-auth.js.map