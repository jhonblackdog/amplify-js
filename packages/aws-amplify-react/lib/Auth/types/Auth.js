"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/**
* Third Party Auth providers
*/
exports.ThirdPartyProvider = {
    Google: 'accounts.google.com',
    Facebook: 'graph.facebook.com'
};
//# sourceMappingURL=Auth.js.map