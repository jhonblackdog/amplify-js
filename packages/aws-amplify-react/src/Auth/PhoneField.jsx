import * as React from 'react';
import { 
    FormField,
    Input,
    InputLabel,
    SelectInput
 } from '../Amplify-UI/Amplify-UI-Components-React';
import AmplifyTheme from '../Amplify-UI/Amplify-UI-Theme';
import countryDialCodes from './common/country-dial-codes.js';
import { I18n } from '@aws-amplify/core';

class PhoneField extends React.Component {
    constructor(props) {
        super(props);
        this.handleInputChange = this.handleInputChange.bind(this);
        this.composePhoneNumber = this.composePhoneNumber.bind(this);

        this.inputs = {
            dial_code: this.props.defaultDialCode || '+1',
            phone_line_number: ''
        }
    }

    composePhoneNumber() {
        return `${this.inputs.dial_code || '+1'}${this.inputs.phone_line_number.replace(/[-()]/g, '')}`;
    }

    handleInputChange(evt) {
        const { name, value } = evt.target;
        this.inputs[name] = value;

        if (this.props.onChangeText) this.props.onChangeText(this.composePhoneNumber());
    }

    render() {
        const { 
            theme=AmplifyTheme, 
            required=true, 
            defaultDialCode='+1', 
            label='Phone Number',
            placeholder='Enter your phone number'
        } = this.props;
        
        return (
            <FormField theme={theme} key="phone_number">
            {
                required ? 
                <InputLabel theme={theme}>{I18n.get(label)} *</InputLabel> :
                <InputLabel theme={theme}>{I18n.get(label)}</InputLabel>
            }
                <SelectInput theme={theme}>
                    <select name="dial_code" defaultValue={defaultDialCode} 
                    onChange={this.handleInputChange}
                    data-test="dial-code-select">
                        {countryDialCodes.map(dialCode =>
                            <option key={dialCode} value={dialCode}>
                                {dialCode}
                            </option>
                        )}
                    </select>
                    <Input
                        placeholder={I18n.get(placeholder)}
                        theme={theme}
                        type="tel"
                        id="phone_line_number"
                        key="phone_line_number"
                        name="phone_line_number"
                        onChange={this.handleInputChange}
                        data-test="phone-number-input"
                    />
                </SelectInput>
            </FormField>
        )
    }
}

export { PhoneField };
export default PhoneField;
