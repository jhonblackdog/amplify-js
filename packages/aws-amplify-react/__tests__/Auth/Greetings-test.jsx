import * as React from 'react';
import { Greetings } from '../../src/Auth/Greetings';

const acceptedStates = [
    'signedIn'
];

const deniedStates = [
    'signIn',  
    'signedUp', 
    'signedOut',
    'forgotPassword',
    'signUp',
    'confirmSignIn',
    'confirmSignUp',
    'verifyContact'
];

describe('Greetings', () => {
    describe('normal case', () => {
        test('render correctly with authState signedIn', () => {
            const wrapper = shallow(<Greetings/>);
            for (let i = 0; i < acceptedStates.length; i += 1){
                wrapper.setProps({
                    authState: acceptedStates[i],
                    authData: {
                        attributes: {
                            name: 'username'
                        }
                    },
                    theme: 'theme'
                });
                expect(wrapper).toMatchSnapshot();
            }
        });

        test('render correctly with hide', () => {
            const wrapper = shallow(<Greetings/>);
            for (let i = 0; i < acceptedStates.length; i += 1){
                wrapper.setProps({
                    authState: acceptedStates[i],
                    theme: 'theme',
                    hide: [Greetings]
                });
                expect(wrapper).toMatchSnapshot();
            }
        });
    });

   
    test('render corrently with other authStates', () => {
        const wrapper = shallow(<Greetings/>);
        
        for (let i = 0; i < deniedStates.length; i += 1){
            wrapper.setProps({
                authState: deniedStates[i],
                theme: 'theme'
            });

            expect(wrapper).toMatchSnapshot();
        }
    });   
});
