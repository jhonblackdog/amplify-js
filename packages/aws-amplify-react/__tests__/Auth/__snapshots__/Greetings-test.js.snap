// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signUp normal case render correctly with authState signedIn 1`] = `
<div
  className="amplify-nav-bar"
  style={
    Object {
      "borderBottom": "1px solid #00fff6",
      "marginBottom": "0.5em",
      "padding": "0.5em 0.5em 0.2em",
    }
  }
>
  <div
    className="amplify-nav-right"
    style={
      Object {
        "fontSize": "0.8em",
        "textAlign": "right",
      }
    }
  >
    Please Sign In / Sign Up
  </div>
</div>
`;

exports[`signUp render corrently with other authStates 1`] = `
<div
  className="amplify-nav-bar"
  style={
    Object {
      "borderBottom": "1px solid #00fff6",
      "marginBottom": "0.5em",
      "padding": "0.5em 0.5em 0.2em",
    }
  }
>
  <div
    className="amplify-nav-right"
    style={
      Object {
        "fontSize": "0.8em",
        "textAlign": "right",
      }
    }
  >
    Please Sign In / Sign Up
  </div>
</div>
`;

exports[`signUp render corrently with other authStates 2`] = `
<div
  className="amplify-nav-bar"
  style={
    Object {
      "borderBottom": "1px solid #00fff6",
      "marginBottom": "0.5em",
      "padding": "0.5em 0.5em 0.2em",
    }
  }
>
  <div
    className="amplify-nav-right"
    style={
      Object {
        "fontSize": "0.8em",
        "textAlign": "right",
      }
    }
  >
    Please Sign In / Sign Up
  </div>
</div>
`;

exports[`signUp render corrently with other authStates 3`] = `
<div
  className="amplify-nav-bar"
  style={
    Object {
      "borderBottom": "1px solid #00fff6",
      "marginBottom": "0.5em",
      "padding": "0.5em 0.5em 0.2em",
    }
  }
>
  <div
    className="amplify-nav-right"
    style={
      Object {
        "fontSize": "0.8em",
        "textAlign": "right",
      }
    }
  >
    Please Sign In / Sign Up
  </div>
</div>
`;

exports[`signUp render corrently with other authStates 4`] = `
<div
  className="amplify-nav-bar"
  style={
    Object {
      "borderBottom": "1px solid #00fff6",
      "marginBottom": "0.5em",
      "padding": "0.5em 0.5em 0.2em",
    }
  }
>
  <div
    className="amplify-nav-right"
    style={
      Object {
        "fontSize": "0.8em",
        "textAlign": "right",
      }
    }
  >
    Please Sign In / Sign Up
  </div>
</div>
`;

exports[`signUp render corrently with other authStates 5`] = `
<div
  className="amplify-nav-bar"
  style={
    Object {
      "borderBottom": "1px solid #00fff6",
      "marginBottom": "0.5em",
      "padding": "0.5em 0.5em 0.2em",
    }
  }
>
  <div
    className="amplify-nav-right"
    style={
      Object {
        "fontSize": "0.8em",
        "textAlign": "right",
      }
    }
  >
    Please Sign In / Sign Up
  </div>
</div>
`;

exports[`signUp render corrently with other authStates 6`] = `
<div
  className="amplify-nav-bar"
  style={
    Object {
      "borderBottom": "1px solid #00fff6",
      "marginBottom": "0.5em",
      "padding": "0.5em 0.5em 0.2em",
    }
  }
>
  <div
    className="amplify-nav-right"
    style={
      Object {
        "fontSize": "0.8em",
        "textAlign": "right",
      }
    }
  >
    Please Sign In / Sign Up
  </div>
</div>
`;

exports[`signUp render corrently with other authStates 7`] = `
<div
  className="amplify-nav-bar"
  style={
    Object {
      "borderBottom": "1px solid #00fff6",
      "marginBottom": "0.5em",
      "padding": "0.5em 0.5em 0.2em",
    }
  }
>
  <div
    className="amplify-nav-right"
    style={
      Object {
        "fontSize": "0.8em",
        "textAlign": "right",
      }
    }
  >
    Please Sign In / Sign Up
  </div>
</div>
`;

exports[`signUp render corrently with other authStates 8`] = `
<div
  className="amplify-nav-bar"
  style={
    Object {
      "borderBottom": "1px solid #00fff6",
      "marginBottom": "0.5em",
      "padding": "0.5em 0.5em 0.2em",
    }
  }
>
  <div
    className="amplify-nav-right"
    style={
      Object {
        "fontSize": "0.8em",
        "textAlign": "right",
      }
    }
  >
    Please Sign In / Sign Up
  </div>
</div>
`;
